connect/README.md

```diff
@@ -0,0 +1,259 @@
+
+{
+  "id": "1",
+  "name": "Mario",
+  "description": "Better description"
+}
+```
+
+hyper-connect also handles the response and returns the result, if the response is not a 2XX response,
+then a rejected promise is returned with the error supplied as the argument in the catch function.
+
+---
+
+## Setup
+
+Before we get started coding, we need to initialize our project. Run the setup script to
+
+* Download and run a hyper instance
+
+``` sh
+cd connect
+./scripts/hyper.sh
+```
+
+In another terminal lets setup our hyper data service:
+
+``` sh
+cd connect
+./scripts/setup.js
+```
+
+In another terminal lets start our API server:
+
+``` sh
+cd connect
+./scripts/start.sh
+```
+
+---
+
+## Update a Character
+
+In the previous workshop, we learned how to add a document to hyper and to retrieve a document from hyper, in this workshop, we will learn how to update a document. The update method for hyper.data takes two arguments, an identifier and the updated document. hyper completely replaces the document, so you must 
+send a full document to hyper. 
+
+> QUESTION: Why does hyper take the entire document to perform an update? This is intentional by design
+> You should keep your documents small and domain focused, this allows for better scale. If you want
+> to learn more about document database design check out this blog post https://blog.hyper.io/document-database-design 
+
+In the `api/update-character.js` file, lets re-write the updateCharacter API handler:
+
+``` js
+import { hyper } from 'https://x.nest.land/hyper-connnect@0.0.7/proxy.js'
+
+export default async function(req, res) {
+  const result = await hyper.data.update(req.params.id, req.body)
+  return res.send(result)
+}
+```
+
+To verify the update is successful, lets use curl to send an update:
+
+In a new terminal window:
+
+``` sh
+curl -X PUT localhost:3000/api/characters/1 \
+-H 'Content-Type: application/json' \
+-d '{"id": "1", "type": "character", "name": "Mario", "description": "updating mario document" }'
+```
+
+---
+
+## Remove a Character
+
+To remove a document from a hyper data service, we will use the `data.remove` method. In `api/remove-character.js` lets re-write the function like so:
+
+``` js
+import { hyper } from 'https://x.nest.land/hyper-connnect@0.0.7/proxy.js'
+
+export default async function(req, res) {
+  const result = await hyper.data.remove(req.params.id)
+  return res.send(result)
+}
+
+```
+
+And we can verify using curl
+
+``` sh
+curl -X DELETE localhost:3000/api/characters/1
+```
+
+---
+
+## List Characters
+
+In order to showcase the list command, we need to seed the database with a bunch of Mario
+Universe characters. Let use the seed script to do this:
+
+``` sh
+./scripts/setup.sh
+```
+
+> NOTE: You can run `./scripts/setup.sh` multiple times in case you get your data out of wack, it will
+> reseed your data service with the characters.json file. Make sure you are in the `connect` folder in
+> your terminal.
+
```

Comment:
> NOTE: You can run `./scripts/setup.sh` multiple times in case you get your data out of whack, it will
> reseed your data service with the characters.json file. Make sure you are in the `connect` folder in
> your terminal.